software-mansion/radon-ide
# Handle eas build locally thorugh buildCommand in launch.json

Fixes #824

`customBuild.buildCommand` assumes the last line of the output is a path to the binary.  This PR changes this behavior to look for all paths in the last line of the output; if it does exist, it takes the first path and extracts the tar file if necessary (this is how iOS is bundled). This seems to be a common use case (especially with eas --local) and, currently, we have no other way of handling `eas build` command locally. In the feature, I believe we should extend `eas.*` launch config to handle `local` flag. 


### How Has This Been Tested: 

- Open Expo 52 Eas from: https://github.com/software-mansion-labs/radon-ide-test-apps/pull/3
- add launch configuration:
```
{
   // Use IntelliSense to learn about possible attributes.
   // Hover to view descriptions of existing attributes.
   // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
   "version": "0.2.0",
   "configurations": [
      {
         "type": "radon-ide",
         "request": "launch",
         "name": "Radon IDE panel",
         "customBuild": {
            "ios": {
               "buildCommand": "npx eas build --platform ios --profile development --local --non-interactive",
            },
            "android": {
               "buildCommand": "npx eas build --platform android --profile development --local --non-interactive",
            }
         },
         "ios": {
            "configuration": "Debug"
         },
         "android": {
            "buildType": "debug"
         }
      }
   ]
}
```
- verify that the app builds (it takes a long time especially on iOS)and IDE works
